chessengine/bitboard.py

```diff
@@ -157,10 +161,17 @@ def add_bitboard_to_repr(board, s, p):
         board_repr = ""
         board_repr += "\u2001" + "\u2001".join(files) + "\n"
         for i in range(8):
+
+            piece_list_str = ""
+            if self.side == "white":
+                piece_list_str = piece_list[8 * i : 8 * i + 8]
+            else:
+                piece_list_str = piece_list[8 * i : 8 * i + 8][::-1]
```

Comment:
Just reversing the `piece_list` list will not be enough. After this patch, the rank and file markers are reversed correctly as expected, but the whole board is not rotated by 180°. This is the current result -

When the board plays black and the user plays white -
![image](https://user-images.githubusercontent.com/64729027/193747709-34d61ee1-8dc4-4fcf-a754-3c5f92c626a8.png)

This is as expected. But when the user plays black, we get -
![image](https://user-images.githubusercontent.com/64729027/193747846-ba2c52ab-be91-4bdd-816b-da66ed76cd5d.png)

The files and ranks are reversed correctly, but in this case the white pieces should be on top and the black pieces should be on the bottom (since the user will expect the pieces they control to be on "their side", i.e. bottom). Here, the white pieces are still on the bottom.

Reply:
I'll fix that after work today.  

😂😂😂 having difficulty seeing difference between white and black pieces because of my terminal.

Reply:
Pushed up the fix and ensured pieces are on the respective side.  

Had to toggle my terminal to have a white background to see accurate depiction of the board.....


![Chess_Black_Perspective](https://user-images.githubusercontent.com/2515617/193973553-c6b48c63-6896-4483-8b10-6edc3b39d350.JPG)
![Chess_White_Perspective](https://user-images.githubusercontent.com/2515617/193973555-9b7c63f9-cccb-4438-a6ec-b67d9e6f9491.JPG)